### PR TITLE
Implement Comments 

### LINKED ISSUE #50: Add Lisp-style line comment syntax
- [ ] Lines starting with `;` should be comments

### PR DESCRIPTION
**Pull request by andrevidela**
fix #50

### CODE REVIEW

File: src/Parse.idr [hunk 1]
```diff
@@ -54,10 +56,26 @@ sizedtok' tok = MkParameters tok (SizedList tok) (const $ pure ())
 Parser' : Type -> Nat -> Type
 Parser' = Parser TPState (sizedtok' Char)
 
+neComments : (Alternative mn, Monad mn, Subset Char (Tok p), Inspect (Toks p) (Tok p), Eq (Tok p)) =>
+           All (Parser mn p ())
```
Comment (epost) [inline src/Parse.idr]: I'm not sure what the conventions are around line breaks and aligning type signatures, (I do think 120 char lines are ok btw) but I think some alignment wrt `:` would be nice. Let's ask about this at the dev meeting tomorrow.

Comment (clayrat) [reply src/Parse.idr]: What I would usually do is to align the `A` on the second line with `(`

File: src/Parse.idr [hunk 2]
```diff
@@ -54,10 +56,26 @@ sizedtok' tok = MkParameters tok (SizedList tok) (const $ pure ())
...
+
+emptyComments : (Alternative mn, Monad mn, Subset Char (Tok p), Inspect (Toks p) (Tok p), Eq (Tok p)) =>
+           All (Parser mn p ())
+emptyComments = cmap () $ string ";\n"
+
+comments : (Alternative mn, Monad mn, Subset Char (Tok p), Inspect (Toks p) (Tok p), Eq (Tok p)) =>
+                All (Parser mn p ())
+comments = emptyComments `alt` neComments
+
+ignorespaces : Alternative mn => Monad mn => Subset Char (Tok p) => Eq (Tok p) =>
```
[the remark this answers is not in the document]
Comment (clayrat) [reply src/Parse.idr]: Also would probably be nice to uncurry all interface constraints for consistency: `(Alternative mn, Monad mn...`

File: src/ParserUtils.idr [hunk 3]
```diff
@@ -0,0 +1,17 @@
+module ParserUtils
```
Comment (clayrat) [inline src/ParserUtils.idr]: This looks general enough to become a PR for TParsec: `except` could go into `Combinators` and `notChar` into `Combinators.Chars`

**Comment by andrevidela** [comment]
@clayrat I fixed the alignment and the casing. Could you take a look and merge if the changes are satisfactory?

File: src/ParserUtils.idr (hunk 3 above)
Comment (andrevidela) [reply src/ParserUtils.idr]: I've opened a pull request (https://github.com/gallais/idris-tparsec/pull/20). I suggest we merge this PR and we change the dependencies ones TParsec is updated. What do you think?

Comment (clayrat) [reply src/ParserUtils.idr]: Yeah, sounds good

**Pull request merged**